definitions/checks/system_registration.rb

```diff
@@ -5,16 +5,26 @@ class Checks::SystemRegistration < ForemanMaintain::Check
     tags :default
 
     confine do
-      file_exists?('/etc/rhsm/rhsm.conf') &&
-        !feature(:foreman_server) &&
-        feature(:foreman_proxy)
+      feature(:downstream) && file_exists?('/etc/rhsm/rhsm.conf')
     end
   end
 
   def run
-    if rhsm_hostname_eql_hostname?
-      warn! 'System is self registered'
-    end
+    assert(
+      !rhsm_hostname_eql_hostname?,
```

Comment:
wouldn't be `rhsm_hostname.downcase != hostname.downcase` more readable?

Reply:
Yes but `rubocop` wouldn't be happy.

Reply:
Thanks for making it a bit better, still I think we should disable this performance cop in the project :) but surely won't block the PR on this. I was checking if it is just me who has problem with the ugly notation and I found in ruby >= 2.4 there is  `str.casecmp?(other_str)` which is better but that we can not use due to the ruby version.

Reply:
+1,  Disabled casecmp performance cop in rubocop file.